walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,78 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> collection = new ArrayList<TimeRange>();
+    collection.add(TimeRange.WHOLE_DAY);
+    for (Event event: events) {
+      splitTimeRange(event.getWhen().start(), event.getWhen().end(), 
+        getTimeRangeWithEvent(event, collection, request), collection);
```

Comment:
nit: put each parameter on a separate line
```suggestion
      splitTimeRange(
          event.getWhen().start(),
          event.getWhen().end(), 
          getTimeRangeWithEvent(event, collection, request),
          collection);
```